src/main/java/Duke.java

```diff
@@ -1,116 +1,138 @@
+        println(Message.LINE.toString());
     }
 
     public static void markTask(int index){
-        println("____________________________________________________________");
+        println(Message.LINE.toString());
         if(index >= taskCount || index < 0){
             println(" Invalid index!");
         } else {
             tasks[index].setDone(true);
             println(" Nice! I've marked this task as done:");
-            printf("   [X] %s\n", tasks[index].getDescription());
+            printf("   [%s][%s] %s\n", tasks[index].getTaskSymbol(), tasks[index].getStatusIcon(), tasks[index]);
         }
-        println("____________________________________________________________");
+        println(Message.LINE.toString());
     }
 
     public static void unmarkTask(int index){
-        println("____________________________________________________________");
+        println(Message.LINE.toString());
         if(index >= taskCount || index < 0){
             println(" Invalid index!");
         } else {
             tasks[index].setDone(false);
             println(" OK, I've marked this task as not done yet:");
-            printf("   [ ] %s\n", tasks[index].getDescription());
+            printf("   [%s][%s] %s\n", tasks[index].getTaskSymbol(), tasks[index].getStatusIcon(), tasks[index]);
         }
-        println("____________________________________________________________");
+        println(Message.LINE.toString());
     }
 
     public static void beginInputLoop(){
         Scanner in = new Scanner(System.in);
+        Parser parser = new Parser();
         while(true){
             try {
                 String input = in.nextLine();
-                String[] args = input.split(" ");
+                parser.parse(input);
                 int index;
-                switch (args[0]) {
+                switch (parser.getCommand()) {
                 case "bye":
+                    in.close();
                     return;
                 case "list":
                     listTasks();
                     break;
                 case "mark":
-                    index = Integer.parseInt(args[1]) - 1;
+                    index = Integer.parseInt(parser.getBody()) - 1;
                     markTask(index);
                     break;
                 case "unmark":
-                    index = Integer.parseInt(args[1]) - 1;
+                    index = Integer.parseInt(parser.getBody()) - 1;
                     unmarkTask(index);
                     break;
+                case "todo":
+                    addTask(new ToDo(parser.getBody()));
+                    break;
+                case "event":
+                    addTask(new Event(parser.getBody(), parser.get("from"), parser.get("to")));
+                    break;
+                case "deadline":
+                    addTask(new Deadline(parser.getBody(), parser.get("by")));
+                    break;
                 default:
```

Comment:
it's nice that the unknown command is the last case, which makes the happy path prominent